packages/instrumentation-browser-navigation/src/instrumentation.ts

```diff
@@ -0,0 +1,448 @@
+        [ATTR_BROWSER_NAVIGATION_HASH_CHANGE]: false,
+      },
+    };
+    this._applyCustomLogRecordData(navLogRecord, this.applyCustomLogRecordData);
+    this.logger.emit(navLogRecord);
+  }
+
+  /**
+   * callback to be executed when using soft navigation
+   */
+  private _onSoftNavigation(
+    changeState: string | null | undefined,
+    navigationEvent?: any
+  ) {
+    const referrerUrl = this._lastUrl;
+    const currentUrl =
+      changeState === 'navigate' && navigationEvent?.destination?.url
+        ? navigationEvent.destination.url
+        : location.href;
+
+    if (referrerUrl === currentUrl) {
+      return;
+    }
+
+    const navType = this._mapChangeStateToType(changeState, navigationEvent);
+    const sameDocument = this._determineSameDocument(
+      changeState,
+      navigationEvent,
+      referrerUrl,
+      currentUrl
+    );
+    const hashChange = this._determineHashChange(
+      changeState,
+      navigationEvent,
+      referrerUrl,
+      currentUrl
+    );
+    const navLogRecord: LogRecord = {
+      eventName: EVENT_NAME,
+      attributes: {
+        [ATTR_URL_FULL]: this._sanitizeUrl(currentUrl),
+        [ATTR_BROWSER_NAVIGATION_SAME_DOCUMENT]: sameDocument,
+        [ATTR_BROWSER_NAVIGATION_HASH_CHANGE]: hashChange,
+        ...(navType ? { [ATTR_BROWSER_NAVIGATION_HASH_TYPE]: navType } : {}),
+      },
+    };
+    this._applyCustomLogRecordData(navLogRecord, this.applyCustomLogRecordData);
+    this.logger.emit(navLogRecord);
+
+    // Update the last known URL after processing
+    this._lastUrl = currentUrl;
+  }
+
+  /**
+   * executes callback {_onDOMContentLoaded } when the page is viewed
+   */
+  private _waitForPageLoad() {
+    // Ensure previous handler is removed before adding a new one
+    if (this._onLoadHandler) {
+      document.removeEventListener('DOMContentLoaded', this._onLoadHandler);
+    }
+    this._onLoadHandler = this._onHardNavigation.bind(this);
+    document.addEventListener('DOMContentLoaded', this._onLoadHandler);
+  }
+
+  /**
+   * implements enable function
+   */
+  override enable() {
+    const cfg = this.getConfig() as BrowserNavigationInstrumentationConfig;
+    const useNavigationApiIfAvailable = !!cfg.useNavigationApiIfAvailable;
+    const navigationApi =
+      useNavigationApiIfAvailable &&
+      ((window as any).navigation as EventTarget);
+
+    // Only patch history API if Navigation API is not available
+    if (!navigationApi) {
+      this._patchHistoryApi();
+    }
+
+    // Always listen for page load
+    this._waitForPageLoad();
+
+    if (navigationApi) {
+      if (this._onNavigateHandler) {
+        navigationApi.removeEventListener('navigate', this._onNavigateHandler);
+        this._onNavigateHandler = undefined;
+      }
+      this._onNavigateHandler = (event: any) => {
+        this._onSoftNavigation('navigate', event);
+      };
+      navigationApi.addEventListener('navigate', this._onNavigateHandler);
```

Comment:
I think we should use the [currententrychange event](https://developer.mozilla.org/en-US/docs/Web/API/Navigation/currententrychange_event) instead. The `navigate` event could be cancelled, while the navigation has been committed when `currententrychange` fires.